src/Bicep.LangServer.IntegrationTests/TelemetryTests.cs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+using Bicep.Core.FileSystem;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.LangServer.IntegrationTests.Helpers;
+using Bicep.LanguageServer.Telemetry;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Window;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Bicep.LangServer.IntegrationTests
+{
+    [TestClass]
+    [SuppressMessage("Style", "VSTHRD200:Use \"Async\" suffix for async methods", Justification = "Test methods do not need to follow this convention.")]
+    public class TelemetryTests
+    {
+        [NotNull]
+        public TestContext? TestContext { get; set; }
+
+        [TestMethod]
+        public async Task VerifyTopLevelDeclarationSnippetInsertionFiresTelemetryEvent()
+        {
+            IDictionary<string, string> properties = new Dictionary<string, string>
+            {
+                { "name", "res-aks-cluster" }
+            };
+
+            TaskCompletionSource<TelemetryEventParams> telemetryReceived = await ResolveCompletionAsync(string.Empty, "res-aks-cluster", TelemetryConstants.EventNames.TopLevelDeclarationSnippetInsertion, properties);
+            TelemetryEventParams telemetryEventParams = await IntegrationTestHelper.WithTimeoutAsync(telemetryReceived.Task);
+            IDictionary<string, object> extensionData = telemetryEventParams.ExtensionData;
+            ICollection<string> keys = extensionData.Keys;
+
+            keys.Count.Should().Be(2);
+            keys.Should().Contain("eventName");
+            keys.Should().Contain("properties");
+            extensionData["eventName"].ToString().Should().Be(TelemetryConstants.EventNames.TopLevelDeclarationSnippetInsertion);
+            extensionData["properties"].ToString().Should().BeEquivalentToIgnoringNewlines(@"{
```

Comment:
[nit] we have JToken assertions - you should be able to use:
```csharp
extensionData["eventName"].Should().DeepEqual(TelemetryConstants.EventNames.TopLevelDeclarationSnippetInsertion);
extensionData["eventName"].Should().DeepEqual(new JObject {
    ["name"] = "res-aks-cluster",
});
```

Reply:
Don't need this anymore. Cleaned up here:
https://github.com/Azure/bicep/commit/62d6763f06b3d53bb8189b517dab0567a37fd0bd